app/static/js/userProfile.js

```diff
@@ -1,323 +1,367 @@
+        },
+        error: function(error) {
+            console.error("An error occurred:", error);
+            // Handle error if needed
+        }
+    });
+  });
+
+
+  function changeAction(e){
+    let profileAction = $(this).val()
+    let username = $(this).data('username')
+    if (profileAction == "Emergency Contact"){
+      window.location.href = `/profile/${username}/emergencyContact`
+    } else if (profileAction == "Insurance Information"){
+      window.location.href = `/profile/${username}/insuranceInfo`
+    } else if(profileAction == "Print Travel Form"){
+      printDocument(`/profile/${username}/travelForm`)
+    } else if (profileAction == "View Service Transcript"){
+      window.location.href = `/profile/${username}/serviceTranscript`
+    } else if (profileAction == "Manage CCE Minor") {
+      window.location.href = `/profile/${username}/cceMinor`
+    }
+    $(this).val('')
+  }
+
+  // This function is to disable all the dates before current date in the ban modal End Date picker
+  $(function(){
+    var banEndDatepicker = $("#banEndDatepicker");
+    banEndDatepicker.datepicker({
+      changeYear: true,
+      changeMonth: true,
+      minDate:+1,
+      dateFormat: "yy-mm-dd",
+    }).attr('readonly','readonly');
+  });
+
+    /*
+     * Ban Functionality
+     */
+
+
+  var programID;
+  $(".ban").click(function() {
+    $.ajax({
+      url: `/profile/${$(this).data("username")}/removeFromTranscript/${$(this).data("programid")}`,
+      type: "GET",
+      success: function(response) {
+          // Check if the program is marked for removal from transcript
+          if (response.removedFromTranscript) {
+              // Handle the case where program is marked for removal
+              $('.removeFromTranscriptCheckbox').prop('checked', true);
+          } else {
+              // Handle the case where program is not marked for removal
+              $('.removeFromTranscriptCheckbox').prop('checked', false);
+          }
+      },
+      error: function(error, status) {
+          console.log(error, status);
+      }
+    });
+  
+    var banButton = $("#banButton")
+    var banEndDateDiv = $("#banEndDate") // Div containing the datepicker in the ban modal
+    var banEndDatepicker = $("#banEndDatepicker") // Datepicker in the ban modal
+    var banNoteDiv = $("#banNoteDiv") // Div containing the note displaying why the user was banned previously
+                                     //Should only diplay when the modal is going to unban a user
+    var banNote = $("#banNote")
+
+    banButton.text($(this).val() + " Volunteer");
+    programID = $(this).data("programid"); // Assign value to programID variable
+    banButton.data("programID", $(this).data("programid"))
+    banButton.data("username", $("#notifyInput").data("username"))
+    banButton.data("banOrUnban", $(this).val());
+    banEndDateDiv.show();
+    banEndDatepicker.val("")
+    $(".modal-title").text($(this).val() + " Volunteer");

```

Comment:
This selector is a problem since there's more than just one modal on the page. You're also changing the titles of all other modals. See the Add Notes Modal under the Bonner Program:
![image](https://github.com/BCStudentSoftwareDevTeam/celts/assets/89316750/ab564eb3-4488-428e-b406-8d6d9fc5a39b)